tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -803,6 +803,14 @@ def setUp(self):
         else:
             self.cleanup_gaia(full_reset=True)
 
+        # Per bug 1156445, check whether the screen is timed out prior to running the test.
+        # When the bootup takes long time, the screen times out when homescreen is opened.
+        if not self.device.is_screen_enabled:
+            self.device.turn_screen_on()
+            if self.device.is_locked:
+                self.device.unlock()
```

Comment:
This change should really be inside `cleanup_gaia`, which already unlocks the device. Is there a cost to calling `turn_screen_on` when the screen is already enabled? If not we could avoid the additional check.